box_error.go

```diff
@@ -0,0 +1,239 @@
+package tarantool
+
+import (
+	"reflect"
+)
+
+const error_extId = 3
+
+// BoxError is a type representing Tarantool `box.error` object: a single
+// MP_ERROR_STACK object with a link to the previous stack error.
+type BoxError struct {
+	Type    string // Type that implies source, for example "ClientError".
+	File    string // Source code file where error was caught.
+	Line    int64  // Line number in source code file.
+	Message string // Text of reason.
+	Errno   int64  // Ordinal number of the error.
+	Errcode int64  // Number of the error as defined in `errcode.h`.
+	// Additional fields depending on error type. For example, if
+	// type is "AccessDeniedError", then it will include "object_type",
+	// "object_name", "access_type".
+	Fields map[interface{}]interface{}
+	Prev   *BoxError // Previous error in stack.
+}
+
+func (err BoxError) Depth() (depth int) {
+	depth = 1
+	for err.Prev != nil {
+		err = *err.Prev
+		depth++
+	}
+	return depth
+}
```

Comment:
1. It would be nice to use this type as common errors. To do this, you need to implement the `error` interface.

https://go.dev/tour/methods/19
```Go
type error interface {
    Error() string
}
```

2. It would be great if the type was field-compatible (in filed-naming) with existing ones:
https://github.com/tarantool/go-tarantool/blob/master/errors.go#L5-L21
`Errcode -> Code`, `Message` -> `Msg`

Reply:
Updated both